tests/test_read_parsers.py

```diff
@@ -483,5 +483,204 @@ def test_clean_seq():
         clean = read.sequence.upper().replace("N", "A")
         assert clean == read.cleaned_seq
 
+
+def test_read_cleaning_consume_seqfile():
+    infile = utils.get_test_data('valid-read-testing.fq')
+
+    x = Counttable(15, int(1e6), 4)
+    x.consume_seqfile(infile)
+
+    # the relevant read will automatically get uppercased => abundance of 2
+    kmer = "caggcgcccaccacc".upper()
+    assert x.get(kmer) == 2
+
+    # the 2nd read with this k-mer in it has an N in it.
+    kmer = "CCTCATCGGCACCAG"
+    assert x.get(kmer) == 1               # this should be 2 in the future
+
+    # the 2nd read with this k-mer in it has a Z in it
+    kmer = "ACTGAGCTTCATGTC"
+    assert x.get(kmer) == 1               # this should be 2 in the future
+
+
+def test_read_cleaning_consume_read_by_read():
+    infile = utils.get_test_data('valid-read-testing.fq')
+
+    x = Counttable(15, int(1e6), 4)
+    for read in ReadParser(infile):
+        x.consume(read.sequence)          # consume raw sequence
+
+    # the relevant read will be entirely ignored
+    # (b/c ReadParser does not uppercase)
+    kmer = "caggcgcccaccacc".upper()
+    assert x.get(kmer) == 1
+
+    # the 2nd read with this k-mer in it has an N in it; 'consume' will ignore.
```

Comment:
I'd rephrase the comment. At the moment I am very puzzled after reading it (ah its being ignored so count should be one, but it is asserted to be two ...??)